pkg/syncutil/cachemanager/cachemanager.go

```diff
@@ -66,8 +213,12 @@ func (c *CacheManager) AddObject(ctx context.Context, instance *unstructured.Uns
 }
 
 func (c *CacheManager) RemoveObject(ctx context.Context, instance *unstructured.Unstructured) error {
-	if _, err := c.opa.RemoveData(ctx, instance); err != nil {
-		return err
+	gvk := instance.GroupVersionKind()
+
+	if c.watchedSet.Contains(gvk) {
+		if _, err := c.opa.RemoveData(ctx, instance); err != nil {
+			return err
+		}
 	}
 
 	// only delete from metrics map if the data removal was succcesful
```

Comment:
Let's compare with the old code:

https://github.com/open-policy-agent/gatekeeper/blob/605dd6339c174654002b27bd6bc57df624879c79/pkg/controller/sync/sync_controller.go#L170-L185

I think our behavior is consistent here. Recall that the opa client being used by the sync_c in the example above is the filtered one, equivalent to our  `if c.watchedSet.Contains(gvk)` check.

As to why, I think it has to do with clean up: A) don't expect this GVK since it's namespace synced, B) don't record metrics for it.